u_quic_frames.go

```diff
@@ -21,6 +21,59 @@ type QUICFrameBuilder interface {
 // It could be used to deterministically build QUIC Frames from crypto data.
 type QUICFrames []QUICFrame
 
+// QUICPacket is a struct that contains a slice of QUICFrame and the total length
+// of all frames including PADDING frames. It can be used to deterministically
+// build QUIC Packets with a target length.
+type QUICPacket struct {
+	Frames QUICFrames
+
+	Length uint16
+}
+
+// Build ingests data from crypto frames without the crypto frame header
+// and returns the byte representation of all frames as specified in
+// the Frames slice. It then calculates the padding sizes needed to
+// reach the specified Length and updates the PADDING frames accordingly.
+func (qp QUICPacket) Build(cryptoData []byte) (payload []byte, err error) {
```

Comment:
```suggestion
func (qp *QUICPacket) Build(cryptoData []byte) (payload []byte, err error) {
```

Let's make `*QUICPacket` instead of `QUICPacket` the one interfacing `QUICFrameBuilder`, just to align with the style of `*QUICRandomFrames`. (`QUICFrames` is an exception as it is a slice instead of a ground-truth struct)